commons-test/src/main/java/org/infinispan/commons/test/PolarionJUnitXMLReporter.java

```diff
@@ -0,0 +1,368 @@
+      }
+      return result;
+   }
+
+   private String replaceAmpersand(String str, Pattern pattern) {
+      int start = 0;
+      int idx = str.indexOf('&', start);
+      if (idx == -1) {
+         return str;
+      }
+      StringBuilder result = new StringBuilder();
+      while (idx != -1) {
+         result.append(str.substring(start, idx));
+         if (pattern.matcher(str.substring(idx)).matches()) {
+            // do nothing it is an entity;
+            result.append("&");
+         } else {
+            result.append("&amp;");
+         }
+         start = idx + 1;
+         idx = str.indexOf('&', start);
+      }
+      result.append(str.substring(start));
+
+      return result.toString();
+   }
+
+   /**
+    * Reset all member variables for next test.
+    */
+   private void resetAll() {
+      m_allTests = Collections.synchronizedList(Lists.<ITestResult>newArrayList());
+      m_numFailed.set(0);
+      m_numSkipped.set(0);
+   }
+
+   private String generateFileName(ISuite suite) {
+      String name = getNameFromSuite(suite);
+      Collection<ISuiteResult> suiteResults = suite.getResults().values();
+      if (suiteResults.size() == 1) {
+         ITestNGMethod[] testMethods = suiteResults.iterator().next().getTestContext().getAllTestMethods();
+         if (testMethods.length > 0) {
+            Class<?> testClass = testMethods[0].getConstructorOrMethod().getDeclaringClass();
+            // If only one test class executed, then use that as the filename
+            String className = testClass.getName();
+            // If only one test package executed, then use that as the filename
+            String packageName = testClass.getPackage().getName();
+            boolean oneTestClass = true;
+            boolean oneTestPackage = true;
+            for (ITestNGMethod method : testMethods) {
+               if (!method.getConstructorOrMethod().getDeclaringClass().getName().equals(className)) {
+                  oneTestClass = false;
+               }
+               if (!method.getConstructorOrMethod().getDeclaringClass().getPackage().getName().equals(packageName)) {
+                  oneTestPackage = false;
+               }
+            }
+            if (oneTestClass) {
+               name = className;
+            } else {
+               if (oneTestPackage) {
+                  name = packageName;
+               }
+            }
+         } else {
+            System.out.println("[" + this.getClass().getSimpleName() + "] " + "Test suite '" + name
+                  + "' results have no test methods");
+         }
+      }
+      return "POLARION-" + name;
+   }
+
+   private String getNameFromSuite(ISuite suite) {
+      String path = suite.getOutputDirectory();
+      int endIndex = path.indexOf(File.separator + "target");
+      if (endIndex != -1) {
+         // Use the project name as the filename
+         if (!System.getenv("MAVEN_PROJECTBASEDIR").equals(System.getProperty("basedir"))) {
```

Comment:
I'm getting a `NullPointerException` here if I run the build from IntelliJ.

TBH I don't see a reason to treat runs from the project directory and from the module directory differently, as the output directory is already different for each module.